File: Python/找完数.py
import sys
t,s=int(sys.argv[1]),0
with open('result.txt','w') as fp:
	for i in range(1,t+1):
		for j in range(1,i):
			if i%j==0 :
				s+=j
		if s==i :
			fp.write(str(i)+' ')
		s=0

File: C/找完数.c
#include<stdio.h>
#include<math.h>
#include<stdlib.h>
int main(int argc,char** argv)
{
	FILE *fp;
	if((fp=fopen("result.txt","w"))==NULL)
		return -1;
	int t,i,j,sum=0;
	t=atoi(argv[1]);
	for(i=1;i<=t;i++)
	{
		for(j=1;j<i;j++)
		{
			if(i%j==0) sum+=j;
		}
		if(sum==i) fprintf(fp,"%d ",i);
		sum=0;
	}
	fclose(fp);
	return 0;
}
